crescent/client.py

```diff
@@ -64,15 +70,30 @@ def __init__(
                 slash commands will be posted globally.
             update_commands:
                 If `True` or not specified, update commands when the bot starts.
+                Only works for gateway-based bots.
             command_hooks:
                 List of hooks to run before all commands.
             command_after_hooks:
                 List of hooks to run after all commands.
         """
-        active_clients[id(app)] = self
 
         self.app = app
 
+        self.is_gateway = isinstance(app, GatewayTraits)
+        if isinstance(app, GatewayTraits):
+            app.event_manager.subscribe(InteractionCreateEvent, self.on_interaction_event)
+
+            if update_commands:
+                app.event_manager.subscribe(StartedEvent, lambda _: self.post_commands())
+        elif isinstance(app, RESTBot):
+            if update_commands:
+                app.add_startup_callback(lambda _: self.post_commands())
+        elif update_commands:
+            raise ValueError(
+                "Crescent cannot update commands automatically for RESTTraits. "
+                "Please pass update_commands=False, or use a RESTBot."
+            )
```

Comment:
I think we just need the `RESTAware` trait to update commands

Reply:
Yes but RESTAware doesn't have any way for us to register a startup callback, and `__init__` isn't async

Reply:
I see. I am not happy with not allowing commands to be updated entirely though. Maybe there could be a function on the client you have to call manually if the doesn't have that callback?

Reply:
Already added that, `client.post_commands`

Reply:
Oh. Can you add how to use `client.post_commands`

Reply:
where?

Reply:
Like 97